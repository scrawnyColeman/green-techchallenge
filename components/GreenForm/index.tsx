import React from "react";
import Button from "../Button";
import GreenLogo from "../GreenLogo";
import Input from "../Input";
import styles from "./assets/scss/GreenForm.module.scss";

const GreenForm: React.FC = (): JSX.Element => {
  return (
    <div className={styles.formWrapper}>
      <div>
        <GreenLogo />
        <h2 className={styles.formTitle}>Operations studio</h2>
        <p className={styles.formSubtitle}>Please enter your email below</p>
      </div>
      <form
        data-testid="form"
        className={styles.form}
        onSubmit={(e) => e.preventDefault()}
      >
        <div>
          <Input type="email" label="Email Address" name="email-login" />
          <Input
            type="checkbox"
            label="Remember this device"
            name="remember-this-device"
          />
        </div>
        <Button
          label="Sign In"
          type="submit"
          onClick={(e) => e.preventDefault()}
        />
      </form>
    </div>
  );
};

export default GreenForm;
