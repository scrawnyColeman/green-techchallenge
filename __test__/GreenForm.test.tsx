import { render, screen, cleanup } from "@testing-library/react";
import GreenForm from "../components/GreenForm";

afterEach(cleanup);

describe("GreenForm", () => {
  it("should render", () => {
    const { asFragment } = render(<GreenForm />);
    expect(asFragment(<GreenForm />)).toMatchSnapshot();
  });
});
