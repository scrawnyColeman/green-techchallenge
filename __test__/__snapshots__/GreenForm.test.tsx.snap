// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GreenForm should render 1`] = `
<DocumentFragment>
  <div
    class="formWrapper"
  >
    <div>
      <img
        alt="Green Logo"
        height="48"
        src="https://green.cdn.energy/branding/logo-r.svg"
        width="96"
      />
      <h2
        class="formTitle"
      >
        Operations studio
      </h2>
      <p
        class="formSubtitle"
      >
        Please enter your email below
      </p>
    </div>
    <form
      class="form"
    >
      <div
        class="inputWrapper"
      >
        <label
          class="inputLabel"
          for="email-login"
        >
          Email Address
        </label>
        <input
          class="textInput"
          id="email-login"
          type="text"
        />
      </div>
      <div
        class="checkboxWrapper"
      >
        <input
          checked=""
          id="remember-this-device"
          type="checkbox"
        />
        <label
          class="checkboxLabel"
          for="remember-this-device"
        >
          Remember this device
        </label>
      </div>
      <button
        aria-label="Sign In"
        class="regularBtn primaryBtn btn"
      >
        Sign In
      </button>
    </form>
  </div>
</DocumentFragment>
`;
